Modules/System/Permission Sets/src/TenantPermissionSubForm.page.al

```diff
@@ -375,6 +644,46 @@ page 9859 "Tenant Permission Subform"
         CurrPage.Update(false);
     end;
 
+    local procedure UpdateSelected(RIMDX: Text[1]; PermissionOption: Option)
+    var
+        TenantPermission: Record "Tenant Permission";
+        OriginalTenantPermission: Record "Tenant Permission";
+    begin
+        CurrPage.SetSelectionFilter(TenantPermission);
+
+        if TenantPermission.FindSet() then
+            repeat
+                case RIMDX of
+                    'R':
+                        if TenantPermission."Object Type" = "Object Type"::"Table Data" then
+                            TenantPermission."Read Permission" := PermissionOption;
+                    'I':
+                        if TenantPermission."Object Type" = "Object Type"::"Table Data" then
+                            TenantPermission."Insert Permission" := PermissionOption;
+                    'M':
+                        if TenantPermission."Object Type" = "Object Type"::"Table Data" then
+                            TenantPermission."Modify Permission" := PermissionOption;
+                    'D':
+                        if TenantPermission."Object Type" = "Object Type"::"Table Data" then
+                            TenantPermission."Delete Permission" := PermissionOption;
+                    'X':
+                        if TenantPermission."Object Type" <> "Object Type"::"Table Data" then
+                            TenantPermission."Execute Permission" := PermissionOption;
+                    '*':
+                        begin
+                            if TenantPermission."Object Type" = "Object Type"::"Table Data" then begin
+                                TenantPermission."Read Permission" := PermissionOption;
+                                TenantPermission."Insert Permission" := PermissionOption;
+                                TenantPermission."Modify Permission" := PermissionOption;
+                                TenantPermission."Delete Permission" := PermissionOption;
+                            end else
+                                TenantPermission."Execute Permission" := PermissionOption;
+                        end;
+                end;
+                TenantPermission.Modify();
```

Comment:
I personally don't understand why we should use a Modify(false) on this place.

Reply:
Perhaps this is a coding style topic. In general, if we already want to avoid modify when not necessary in order to save SQL time we might want to optimize further and do a simple modify without running. I found it very useful in posting routines. In this particular case there is not code on Modify trigger therefore the impact is minor anyway. My suggestion is to do as you suggest if there is a consensus that this is the best practice.